srtcore/core.cpp

```diff
@@ -4987,8 +4979,8 @@ void CUDT::acceptAndRespond(const sockaddr *peer, CHandShake *hs, const CPacket
     }
 
     // get local IP address and send the peer its IP address (because UDP cannot get local IP address)
-    memcpy(m_piSelfIP, hs->m_piPeerIP, 16);
-    CIPAddress::ntop(peer, hs->m_piPeerIP, m_iIPversion);
+    memcpy(m_piSelfIP, hs->m_piPeerIP, sizeof m_piSelfIP);
+    CIPAddress::ntop(peer, hs->m_piPeerIP);
```

Comment:
Line 4173

```
void CUDT::applyResponseSettings()
{
    ...
    memcpy(m_piSelfIP, m_ConnRes.m_piPeerIP, 16);
```
replace with
```
memcpy(m_piSelfIP, m_ConnRes.m_piPeerIP, sizeof m_piSelfIP);
```

Reply:
Good point, but there's lot to be changed in here in order to align it to the code conventions.